nxdrive/gui/view.py

```diff
@@ -355,9 +357,11 @@ def setData(self, index: QModelIndex, value: Any, role: int = None) -> None:
     @pyqtSlot(dict)
     def set_progress(self, action: Dict[str, Any]) -> None:
         for i, item in enumerate(self.items):
-            if (item["engine"], item["name"]) != (action["engine"], action["name"]):
+            if (item["engine"], item["doc_pair"]) != (
+                action["engine"],
+                action["doc_pair"],
+            ):
```

Comment:
Outh, this is ugly :O

Reply:
Very XD

Reply:
Should we do it in two times? It will also slightly improve the checks by not constructing a tuple each time. Something like:
```python
if (
    item["engine"] != action["engine"]
    or item["doc_pair"] != action["doc_pair"]
):
```